### PR TITLE
feat: Enable  FPL Release 19.6 28/10 CCD definitions

### PR DESCRIPTION
**Pull request by samuelzerozero**
**Before creating a pull request make sure that:**

- [ x] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [ x] README and other documentation has been updated / added (if needed)
- [ x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-2354

### Change description ###
Enable PI18 CCD definitions for release

feat: upload draft cmo (FPLA-2019) (#1775)

feat: hearing adjournment and re-listing (FPLA-1916) (#1760)

feat: automate sending of notice of proceedings (FPLA-1754) (#1748)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by Jamesnelson117 — COMMENTED** [review]
I'm happy with the changes made for automate notice of proceedings.

**Review by tomaszpowroznik — APPROVED** [review]

File: ccd-definition/AuthorisationCaseEvent/CareSupervision/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -1033,6 +1033,20 @@
...
+    "UserRole": "caseworker-publiclaw-gatekeeper",
+    "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "CaseEventID": "createNoticeOfProceedings",
+    "UserRole": "caseworker-publiclaw-courtadmin",
+    "CRUD": "R"
+  },
```
Comment (tomaszpowroznik) [inline ccd-definition/AuthorisationCaseEvent/CareSupervision/AuthorisationCaseEvent.json:1049]: probably should go into DEPRECATED file

File: ccd-definition/CaseField/CareSupervision/CMO-prod.json [hunk 2]
```diff
@@ -1,59 +0,0 @@
...
-    "CaseTypeID": "CARE_SUPERVISION_EPO",
-    "ID": "showHearingsMultiTextArea",
-    "Label": "Field to store conditional result for showing multiHearingsWithCMOs",
-    "FieldType": "Text",
-    "SecurityClassification": "Public"
-  },
-  {
-    "LiveFrom": "01/01/2017",
-    "CaseTypeID": "CARE_SUPERVISION_EPO",
-    "ID": "showHearingsSingleTextArea",
```
Comment (tomaszpowroznik) [inline ccd-definition/CaseField/CareSupervision/CMO-prod.json:54]: is it safe to delete these fields completely? If some prod case already contains any of this filed we will get exception from ccd that no definition present

Comment (samuelzerozero) [reply ccd-definition/CaseField/CareSupervision/CMO-prod.json:54]: this is one of the transient fields 
    public static String[] transientFields() {
        return new String[]{
            "showCMOsSentToJudge", "cmosSentToJudge", "cmoUploadType", "pastHearingsForCMO", "futureHearingsForCMO",
            "cmoHearingInfo", "showReplacementCMO", "previousCMO", "uploadedCaseManagementOrder", "replacementCMO",
            "cmoSupportingDocs", "cmoJudgeInfo", "cmoToSend",
            // TODO: 19/10/2020 Delete after this comment when FPLA-2019 is toggled on
            "numHearingsWithoutCMO", "singleHearingWithCMO", "multiHearingsWithCMOs", "showHearingsSingleTextArea",
            "showHearingsMultiTextArea"
        };
    }
@tomaszpowroznik 
do you reckon that there's a high chance to get that when the event is in flight?

Comment (samuelzerozero) [reply ccd-definition/CaseField/CareSupervision/CMO-prod.json]: Just added CMO-deprecated with the transient deprecated fields

File: ccd-definition/AuthorisationCaseEvent/CareSupervision/AuthorisationCaseEvent.json (hunk 1 above)
Comment (samuelzerozero) [reply ccd-definition/AuthorisationCaseEvent/CareSupervision/AuthorisationCaseEvent.json:1049]: would in this case have the two rules (one with  "CRUD": "R" and  the other with "CRUD": "CRU") transformed?

Comment (tomaszpowroznik) [reply ccd-definition/AuthorisationCaseEvent/CareSupervision/AuthorisationCaseEvent.json:1049]: both should have R

File: ccd-definition/CaseField/CareSupervision/CMO-prod.json (hunk 2 above)
Comment (tomaszpowroznik) [reply ccd-definition/CaseField/CareSupervision/CMO-prod.json]: if they are temporary thats fine, but just checked and noHearingsForCMOs is not removed in controller

**Review by tomaszpowroznik — APPROVED** [review]

**Pull request merged**